app/helpers/record_helper.rb

```diff
@@ -190,11 +200,18 @@ def availability_action
     elsif relevant_fulltext_link?(@record.fulltext_link)
       'availability_full'
 
+    # Sometimes sfx links show up elsewhere and are really good links.
+    elsif check_custom_link?
+      'availability_check_online'
+
     # When all else fails, make SFX sort it out.
     else
       @sfx_link = SFXHandler.new(
         title: @record.eds_title,
-        doc_number: clean_an,
+        source_title: @record.source_title,
+        year: @record.eds_publication_year,
+        volume: @record.eds_volume,
+        doc_number: clean_an
```

Comment:
I think I've got a handle on the rest of this PR, but I'm looking at the addition of `year` and `volume` to this, and assuming that I'm missing something. I see them already in the SFXHandler, so are we just starting to pass them from here (while they're already passed from other places that invoke SFXHandler) ?

Reply:
SFXHandler uses these values if we pass them in... but we weren't paying them in here so it wasn't using them. We should have been passing them in as they are useful bits of information.